### DISCUSSION TITLE
Plugin Initialization Failed

### DISCUSSION THREAD

**Discussion by diklein** [potmat/homebridge-google-nest-sdm#190] [Q&A]
Hi! I'm struggling to get everything working. I'm currently seeing "Plugin initialization failed, there was a failure with event subscription" in the logs along with "Failed to retrieve auth metadata with error: invalid_client."

I was able to create a pull subscription and I see it in Google Cloud. However, I am unable to complete the "initiate events" step. I make the following call:

```
curl -X GET 'https://smartdevicemanagement.googleapis.com/v1/enterprises/project-id/devices' \
    -H 'Content-Type: application/json' \
    -H 'Authorization: Bearer access-token'
```

And I get this response:

```
"error": {
"code": 401,
"message": "Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",
"status": "UNAUTHENTICATED"
}
```

Am I doing something wrong? Is this step actually necessary?

**Comment by ghost** [comment]
Make sure to enter the `project-id` in the GET URL to the Nest project URL, and the `access-token` value to the OAUTH key that you received from the [_Link your account_ section](https://developers.google.com/nest/device-access/authorize#link_your_account). It should look something like this (in macOS terminal) prior to sending the curl (greyed out for privacy)

<img width="1129" alt="Screenshot 2024-12-31 at 16 11 56" src="https://github.com/user-attachments/assets/151b6b0a-9324-45bb-90ea-659cf8725b6f" />

**Reply by diklein** [reply]
Oh of course. I removed them just for the post.

**Reply by ghost** [reply]
Oh sorry, I totally forgot that this didn't work for me either (thought this was another earlier step I was having difficulties getting going). This terminal screenshot is from one of my failed attempts 🤣

I did add in the gcpProjectId into the config and everything seems to be working fine for me... no weird log entries/etc.
